pageserver/src/tenant.rs

```diff
@@ -261,7 +261,8 @@ impl UninitializedTimeline<'_> {
             .await
             .context("Failed to flush after basebackup import")?;
 
-        self.initialize(ctx)
+        let mut timelines = self.owning_tenant.timelines.lock().unwrap();
+        self.initialize_with_lock(&mut timelines, false, true)
```

Comment:
As I noticed several times this PR is not just adding check for duplicates.
This check actually detects errors.
One of them is related with importing file fro backup. Right now it cause loading each layer twice.
I have to pass `load_timeline=false` to `initialize_with_lock` to prevent duplication.

Reply:
Not loading it twice is of course much better. I wonder if you added the comment to the right place, because right now it's within `fn bootstrap_timeline` and not here?

Reply:
Suggested comment:
```suggestion
        // Initialize without loading the layer map. We started with an empty layer map, and already
        // updated it for the layers that we created during the import.
        let mut timelines = self.owning_tenant.timelines.lock().unwrap();
        self.initialize_with_lock(&mut timelines, false, true)
```

Reply:
Done